### ISSUE TITLE
Whitespace in vm name not handled correctly

### ISSUE THREAD

**Issue by fenjas** [vmkdaily/vFlux-Stats-Kit#3]
Great work to begin with. I noticed that any vm name that has whitespace in it, will cause cURL to throw an error, example:

{"error":"unable to parse 'net.usage.average,host=VMware 5.5 (Simon),type=VM,vc=192.168.xxx.xxx,cluster=host,instance=vmnic1,unit=KBps,interval=20,numcpu=2,memorygb=4 value=45 1475680910257178880': invalid field format"}

My quick fix is to replace any whitespace with an underscore as follows;

$measurement = $stat.MetricId
$value = $stat.Value
$name = $vm.Name
$name = $name.replace(' ','_')

This worked for me though I haven't tried playing around with curl's syntax.

**Comment by vmkdaily** [comment]
Thanks much!  Great find!

**Comment by fenjas** [comment]
No prob.

**Comment by PaulieShore** [comment]
Ran into the same problem. Thanks for the solution. We have numerous clusters and data center names which contain spaces. However Replace doesn't seem to work with Cluster names. 

What I've tried:
$cluster = $vm.VMHost.Parent
$cluster = $cluster.Replace(" ","_")

Returns:
Method invocation failed because
[VMware.VimAutomation.ViCore.Impl.V1.Inventory.ClusterImpl] does not contain a method named 'Replace'.

Any suggestions? Thanks!

**Comment by PaulieShore** [comment]
I'm no programmer but I was able to figure out $cluster is an array. I wasn't able to easily modify $cluster.Name as it's read-only. As far as I can tell, the $cluster variable is simply to grab the name of the cluster. Please correct me if I'm wrong. To get around this issue I simply overwrote the array with the correct variable. This is probably dirty but it worked. Not sure if I broke anything else... 

		$cluster = $vm.VMHost.Parent
		$clusterName = $cluster.Name.Replace(" ","_")
		$cluster = $clusterName

**Comment by vmkdaily** [comment]
The scripts have been updated to support spaces in the virtual machine DisplayName and vSphere Cluster name for returned data points.  There are no other datapoints that can have a space (i.e. vmhost name is restricted).

If one day, we add portgroup names (something that can be named by users) or blue folders, then we can handle similarly.  For now it's just vm and cluster that were impacted and are now resolved.

By default we set to a real space that is understood by InfluxDB now.  So we do a replace of `' '` with the  `'\ '` .  If desired, set to `'_'`.  See the preferences section of the script for this setting.

Thanks to all for the feedback!
